util/env_posix.cc

```diff
@@ -305,15 +306,19 @@ class PosixEnv : public Env {
     }
     if (options.use_mmap_writes && !forceMmapOff_) {
       result->reset(new PosixMmapFile(fname, fd, page_size_, options));
-    } else if (options.use_direct_writes && !options.use_mmap_writes) {
+    } else if (use_direct) {
 #ifdef OS_MACOSX
       if (fcntl(fd, F_NOCACHE, 1) == -1) {
         close(fd);
         s = IOError(fname, errno);
         return s;
       }
 #endif
-      result->reset(new PosixWritableFile(fname, fd, options));
+      if (fname.rfind(".log") != std::string::npos) {
+        result->reset(new WALWritableFile(fname, fd, options));
+      } else {
```

Comment:
I think there is no need to add a new WritableFile Class for WAL. Would it be better to add a new param - `FileType` here and change the implemenation of `PosixWritableFile`?